addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +442,34 @@ function factory(dependencies) {
                 this.originThread.isModeratedByCurrentPartner
             );
         }
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsSubjectEqualThreadName() {
+            if (!this.subject || !this.originThreadName) {
+                return false;
+            }
+
+            function cleanSubject (prefix, threadName, cleanedSubject) {
+                for (const pref of prefix) {
+                    if (threadName === cleanedSubject) {
+                        return true;
+                    }
+                    if (cleanedSubject.startsWith(pref)) {
+                        cleanedSubject = cleanedSubject.replace(pref, '').trim();
+                        prefix = prefix.filter((pr) => pr !== pref);
+                        return cleanSubject(prefix, threadName, cleanedSubject);
+                    }
+                }
+                return threadName === cleanedSubject;
+            }
+
+            const prefix = ['re:', 'fw:', 'fwd:'];
+            let cleanedSubject = this.subject.toLowerCase();
+            const threadName = this.originThreadName.toLowerCase().trim();
+            return cleanSubject(prefix, threadName, cleanedSubject);
```

Comment:
I would remove the recursion to use a loop in this case, it would be more readable.

Something like that in pseudo code:
```
starting subject = subject
cleaned subject (= undefined)
while starting subject != cleaned subject:
   cleaned subject = starting subject
   for prefix in prefix_list
       if subject start with prefix
           cleaned subject = replace
           break
   if cleaned subject === thread name
       return true
```